src/Microsoft.Azure.WebJobs.ServiceBus/EventHubs/EventHubConfiguration.cs

```diff
@@ -25,8 +23,11 @@ public class EventHubConfiguration : IExtensionConfigProvider
         // Event Hub Names are case-insensitive.
         // The same path can have multiple connection strings with different permissions (sending and receiving), 
         // so we track senders and receivers separately and infer which one to use based on the EventHub (sender) vs. EventHubTrigger (receiver) attribute. 
-        // Connection strings may also encapsulate different endpoints. 
-        private readonly Dictionary<string, EventHubClient> _senders = new Dictionary<string, EventHubClient>(StringComparer.OrdinalIgnoreCase);
+        // Connection strings may also encapsulate different endpoints.
+        //
+        // The client cache must be thread safe because clients are accessed/added on the function
+        // invocation path (BuildFromAttribute)
+        private readonly ConcurrentDictionary<string, EventHubClient> _clients = new ConcurrentDictionary<string, EventHubClient>(StringComparer.OrdinalIgnoreCase);
         private readonly Dictionary<string, ReceiverCreds> _receiverCreds = new Dictionary<string, ReceiverCreds>(StringComparer.OrdinalIgnoreCase);
```

Comment:
The other two dictionaries are used on the indexing path during listener startup, where there is no concurrency